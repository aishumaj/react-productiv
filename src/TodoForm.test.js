import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoForm from "./TodoForm"

let mock = jest.fn();

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<TodoForm handleSave={mock} />);
  });

  it("contains expected form and sections", function () {
    const { container} = render(<TodoForm handleSave={mock}/>);
    expect(container.querySelector(".NewTodoForm")).toBeInTheDocument();
    expect(container.querySelectorAll(".mb-3").length).toEqual(3);
    expect(container.querySelector("button")).toContainHTML("GÃ¸!");
  });
  
  it("matches initial form snapshot", function () {
    const { asFragment } = render(<TodoForm handleSave={mock}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("matches data-entered snapshot", function () {
    const { container } = render(<TodoForm handleSave={mock}/>);
  
    const titleInput = container.querySelector("[name=title]");
    const descriptionInput = container.querySelector("[name=description]");
    const priorityInput = container.querySelector("[name=priority]");
  
    fireEvent.input(titleInput, { target: { value: "changing" } });
    fireEvent.input(descriptionInput, { target: { value: "description here" } });
    fireEvent.input(priorityInput, { target: { value: "2" } });
  
    expect(container).toMatchSnapshot();
  });
  
  it("submitting form works", function () {
    const { container } = render(<TodoForm handleSave={mock}/>);
  
    const titleInput = container.querySelector("[name=title]");
    const descriptionInput = container.querySelector("[name=description]");
    const priorityInput = container.querySelector("[name=priority]");
  
    fireEvent.input(titleInput, { target: { value: "changing" } });
    fireEvent.input(descriptionInput, { target: { value: "description here" } });
    fireEvent.input(priorityInput, { target: { value: "2" } });
  
    expect(mock).toHaveBeenCalledTimes(0);
  
    fireEvent.click(container.querySelector(".NewTodoForm-addBtn"))
    expect(mock).toHaveBeenCalledTimes(1);
  
    // expect all form inputs to be empty
  
    expect(container.querySelectorAll("input[value='']")).toHaveLength(1);
    expect(container.querySelector("textarea")).toContainHTML("");
  });

});