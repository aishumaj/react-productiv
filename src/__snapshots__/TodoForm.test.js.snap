// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches data-entered snapshot 1`] = `
<div>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value="changing"
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      >
        description here
      </textarea>
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</div>
`;

exports[`productiv app matches initial form snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      />
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`productiv app matches snapshot 1`] = `
<div>
  <form
    class="NewTodoForm"
  >
    <div
      class="mb-3"
    >
      <input
        aria-label="Title"
        class="form-control"
        id="newTodo-title"
        name="title"
        placeholder="Title"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <textarea
        aria-label="Description"
        class="form-control"
        id="newTodo-description"
        name="description"
        placeholder="Description"
      />
    </div>
    <div
      class="mb-3 d-flex justify-content-between"
    >
      <div
        class="w-75 d-flex justify-content-between"
      >
        <label
          class="d-inline-flex"
          for="newTodo-priority"
        >
          Priority:  
        </label>
        <select
          class="form-control form-control-sm d-inline-flex"
          id="newTodo-priority"
          name="priority"
        >
          <option
            value="1"
          >
            Ultra-Über
          </option>
          <option
            value="2"
          >
            Über
          </option>
          <option
            value="3"
          >
            Meh
          </option>
        </select>
      </div>
      <button
        class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
      >
        Gø!
      </button>
    </div>
  </form>
</div>
`;
